dftext.py

```diff
@@ -74,20 +78,22 @@ def decode(self, in_text, index=False):
             decompressed = decompressed[6:]
             if record_length != record_length_2:
                 raise ParserDecodeError('Record lengths do not match')
-            record, decompressed = decompressed[:record_length], decompressed[record_length:]
+            record = decompressed[:record_length]
+            decompressed = decompressed[record_length:]
             if index:
                 record = self.index_scramble(record)
             records.append(record)
-        return '\n'.join(records) + '\n'
+        return convert(str, b'\n'.join(records) + b'\n')
 
     def decode_file(self, in_file, *args, **kwargs):
         return self.decode(self.read_file(in_file), *args, **kwargs)
 
     def encode(self, in_text, index=False, level=zlib.Z_DEFAULT_COMPRESSION):
-        records = in_text.rstrip('\n').split('\n')
-        out_text = ''.join([struct.pack('<LH', len(record), len(record))
-                            + (self.index_scramble(record) if index else record)
-                            for record in records])
+        records = [s.strip() for s in in_text.strip(b'\n').split(b'\n')]
```

Comment:
Is the `s.strip()` intended to get rid of `\r` characters? I'm pretty sure splitlines() would work too:
```
records = in_text.rstrip('\n').splitlines()
```
(although I'm not sure if this will work without other changes for encoding in Python 3)